dashboards/resource_lakeview_dashboard.go

```diff
@@ -0,0 +1,233 @@
+	var content []byte
+	if filePath != "" {
+		content, err = readFileContent(filePath)
+		if err != nil {
+			return "", "", err
+		}
+	} else {
+		log.Printf("[INFO] Reading `serialized_json` of %d bytes", len(jsonStr))
+		content = []byte(jsonStr)
+	}
+	md5Hash = calculateMd5Hash(content)
+	return string(content), md5Hash, nil
+}
+
+// ResourceLakeviewDashboard manages lakeview dashboards
+func ResourceLakeviewDashboard() common.Resource {
+	s := common.StructToSchema(dashboards.Dashboard{},
+		func(s map[string]*schema.Schema) map[string]*schema.Schema {
+			s["create_time"].Computed = true
+			s["dashboard_id"].Computed = true
+			s["display_name"].Optional = false
+			s["display_name"].Required = true
+			s["etag"].Computed = true
+			s["lifecycle_state"].Computed = true
+			s["parent_path"].Optional = false
+			s["parent_path"].Required = true
+			s["parent_path"].ForceNew = true
+			s["path"].Computed = true
+			s["serialized_dashboard"].Optional = true
+			s["serialized_dashboard"].ConflictsWith = []string{"file_path"}
+			s["serialized_dashboard"].DiffSuppressFunc = func(k, old, new string, d *schema.ResourceData) bool {
+				_, ok := d.GetOk("file_path")
+				if ok {
+					return true
+				}
+				_, new_json_hash, _ := readSerializedJsonContent(new, "")
+				// fmt.Println(new_json_hash, d.Get("md5").(string))
+				// fmt.Println(new, d.Get("serialized_dashboard").(string), old)
+				return d.Get("md5").(string) == new_json_hash && !d.Get("dashboard_change_detected").(bool)
+				// return !d.Get("dashboard_change_detected").(bool)
+			}
+			s["update_time"].Computed = true
+			s["warehouse_id"].Optional = false
+			s["warehouse_id"].Required = true
+
+			s["file_path"] = &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"serialized_dashboard"},
+			}
+			s["md5"] = &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			}
+			s["dashboard_change_detected"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			}
+			s["embed_credentials"] = &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			}
+			return s
+		})
+	return common.Resource{
+		Schema: s,
+		Create: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
+			w, err := c.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			var new_dashboard dashboards.CreateDashboardRequest
+			common.DataToStructPointer(d, s, &new_dashboard)
+			content, md5Hash, err := readSerializedJsonContent(d.Get("serialized_dashboard").(string), d.Get("file_path").(string))
+			if err != nil {
+				return err
+			}
+			d.Set("md5", md5Hash)
+			new_dashboard.SerializedDashboard = content
+			created_dashboard, err := w.Lakeview.Create(ctx, new_dashboard)
+			if err != nil {
+				return err
+			}
+
+			time.Sleep(2 * time.Second)
```

Comment:
Why do we need to sleep for 2 seconds here?

Reply:
I removed it now. It was a hack for when API was not working correctly